src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullCoalescingOperator.cs

```diff
@@ -180,6 +189,23 @@ private bool IsStringConcat(BoundExpression expression)
             return false;
         }
 
+        private bool IsDefaultValueOrConvertedDefaultValue(BoundExpression expression)
+        {
+            if (expression.IsDefaultValue())
+            {
+                return true;
+            }
+
+            if (expression.Kind == BoundKind.Conversion)
```

Comment:
>if [](start = 12, length = 2)

It's not a big difference, but I think this would be more readable as: 1) loop to remove identity conversions, 2) check if the operand is a default value.

Reply:
Done